### PR TITLE
chore(core):removed unused generics, added extra ones

### PR DESCRIPTION
**Pull request by Glazomer**
Issue #546

In core.ts:5 
Rewritten getDisplayName function — removed unused generics.

### CODE REVIEW

File: packages/core/core.ts [hunk 1]
```diff
@@ -70,7 +67,6 @@ export function withBemMod<T, U extends IClassNameProps = {}>(
 
     function BemMod(props: T & K) {
       modNames = modNames || Object.keys(mod)
-      // TODO: For performance can rewrite `every` to `for (;;)`.
```
Comment (yarastqt) [inline packages/core/core.ts]: This PR not about this, let's returns this comment

File: packages/core/core.ts [hunk 2]
```diff
@@ -2,11 +2,8 @@ import { ComponentType, StatelessComponent, createElement } from 'react'
 import { cn, NoStrictEntityMods, ClassNameFormatter } from '@bem-react/classname'
 import { classnames } from '@bem-react/classnames'
 
-function getDisplayName<T>(Component: ComponentType<T> | string) {
-  return typeof Component === 'string'
-    ? Component
-    : Component.displayName || Component.name || 'Component'
-}
+const getDisplayName = (Component: ComponentType | string) =>
```
Comment (yarastqt) [inline packages/core/core.ts]: Why u rewrite for function expression? All code have function declaration

File: packages/core/test/withBemMod.test.tsx [hunk 3]
```diff
@@ -17,7 +17,7 @@ interface IPresenterProps extends IClassNameProps {
 
 const presenter = cn('Presenter')
 
-const Presenter: React.FC<IPresenterProps> = ({ className }) => (
+const Presenter: React.FC<IPresenterProps> = ({ className }: any) => (
```
[the remark this answers is not in the document]
Comment (Glazomer) [reply packages/core/test/withBemMod.test.tsx]: It does, but TSLinter gives warnings, so if there no difference

File: packages/core/core.ts (hunk 1 above)
Comment (Glazomer) [reply packages/core/core.ts]: agree, sorry

File: packages/core/core.ts (hunk 2 above)
Comment (Glazomer) [reply packages/core/core.ts]: Agree, will make it back to FD

**Pull request closed without merging**